### PR TITLE
Remove schema migration scripts

### LINKED ISSUE #50: Schema migration error handling
## Error handling

When a schema is updated and an error occurs on a node migration, no error is logged.

Also the migration aborts resulting in some nodes migrated and some not. The nodes which are not migrated can't be updated by a schema update anymore.
TODO: Create an endpoint which migrates all nodes with older schema version to the most recent schema version.

It can happen that a migration fails due to webroot uniqueness checks or script errors. It must be possible to fix the script error and retry the migration.

### Option A

Wait for Versioning and rollback the entire migration. This way a new migration can be executed.

### Option B
Update a specific change script and execute the migration again.
Note: This may not fix errors which occur due to uniqueness checks.

### Extra
It would be good to lock nodes which have been migrated by a migration that has not yet been completed. Additionally it should not be possible to create new nodes using a schema which has been created by a migration which has not yet been completed.

### PR DESCRIPTION
**Pull request by philippguertler**
Resolves #50

### CODE REVIEW

**Comment by Jotschi** [comment]
buildnow

File: core/src/main/java/com/gentics/mesh/core/data/schema/impl/FieldTypeChangeImpl.java [hunk 1]
```diff
@@ -130,9 +152,187 @@ public FieldSchemaContainer apply(FieldSchemaContainer container) {
...
+				return Collections.singletonMap(getFieldName(), changeToNode(oldSchema, oldContent));
+			default:
+				throw error(BAD_REQUEST, "Unknown type {" + newType + "} for change " + getUuid());
+		}
+	}
+
+	private BooleanField changeToBoolean(FieldSchemaContainer oldSchema, FieldContainer oldContent) {
+		String fieldName = getFieldName();
+		FieldSchema fieldSchema = oldSchema.getField(fieldName);
+		TypeConverter typeConverter = new TypeConverter();
```
Comment (Jotschi) [inline core/src/main/java/com/gentics/mesh/core/data/schema/impl/FieldTypeChangeImpl.java]: Make typeconverter static

File: core/src/test/java/com/gentics/mesh/core/schema/field/BooleanFieldMigrationTest.java [hunk 2]
```diff
@@ -34,20 +32,6 @@ public void testRemove() throws Exception {
 		removeField(CREATEBOOLEAN, FILLTRUE, FETCH);
 	}
 
-	@Test
-	@Override
-	public void testRename() throws Exception {
```
Comment (Jotschi) [inline core/src/test/java/com/gentics/mesh/core/schema/field/BooleanFieldMigrationTest.java]: Check why this case was removed?

File: core/src/test/java/com/gentics/mesh/core/schema/field/BooleanListFieldMigrationTest.java [hunk 3]
```diff
@@ -36,15 +32,6 @@ public void testRemove() throws Exception {
 		removeField(CREATEBOOLEANLIST, FILL, FETCH);
 	}
 
-	@Override
-	@Test
-	public void testRename() throws Exception {
```
Comment (Jotschi) [inline core/src/test/java/com/gentics/mesh/core/schema/field/BooleanListFieldMigrationTest.java]: Check why this was removed.

File: core/src/test/java/com/gentics/mesh/core/schema/field/DateFieldMigrationTest.java [hunk 4]
```diff
@@ -35,15 +33,6 @@ public void testRemove() throws Exception {
 		removeField(CREATEDATE, FILL, FETCH);
 	}
 
-	@Override
-	@Test
-	public void testRename() throws Exception {
```
Comment (Jotschi) [inline core/src/test/java/com/gentics/mesh/core/schema/field/DateFieldMigrationTest.java]: Check why this was removed.

File: core/src/test/java/com/gentics/mesh/core/schema/field/DateListFieldMigrationTest.java [hunk 5]
```diff
@@ -51,15 +48,6 @@ public void testRemove() throws Exception {
 		removeField(CREATEDATELIST, FILL, FETCH);
 	}
 
-	@Override
-	@Test
-	public void testRename() throws Exception {
```
Comment (Jotschi) [inline core/src/test/java/com/gentics/mesh/core/schema/field/DateListFieldMigrationTest.java]: Check why this was removed.

File: core/src/test/java/com/gentics/mesh/core/schema/field/FieldMigrationTestcases.java [hunk 6]
```diff
@@ -11,13 +11,6 @@
 	 */
 	void testRemove() throws Exception;
 
-	/**
-	 * Test renaming the field
-	 * 
-	 * @throws Exception
-	 */
-	void testRename() throws Exception;
```
Comment (Jotschi) [inline core/src/test/java/com/gentics/mesh/core/schema/field/FieldMigrationTestcases.java]: Check why this was removed.

File: core/src/test/java/com/gentics/mesh/core/schema/field/HtmlFieldMigrationTest.java [hunk 7]
```diff
@@ -43,15 +40,6 @@ public void testRemove() throws Exception {
 		removeField(CREATEHTML, FILLTEXT, FETCH);
 	}
 
-	@Test
-	@Override
-	public void testRename() throws Exception {
```
Comment (Jotschi) [inline core/src/test/java/com/gentics/mesh/core/schema/field/HtmlFieldMigrationTest.java]: Check why this was removed.

File: rest-model/src/main/java/com/gentics/mesh/core/rest/node/FieldMap.java [hunk 8]
```diff
@@ -40,6 +40,14 @@
 	 */
 	Field put(String fieldKey, Field field);
 
+	default Field put(Map.Entry<String, Field> entry) {
```
Comment (Jotschi) [inline rest-model/src/main/java/com/gentics/mesh/core/rest/node/FieldMap.java]: Javadoc

File: rest-model/src/main/java/com/gentics/mesh/core/rest/node/FieldMap.java [hunk 9]
```diff
@@ -255,4 +263,5 @@
 	@JsonIgnore
 	Set<String> getUrlFieldValues(Schema schema);
 
+	void clear();
```
Comment (Jotschi) [inline rest-model/src/main/java/com/gentics/mesh/core/rest/node/FieldMap.java]: Javadoc

File: rest-model/src/main/java/com/gentics/mesh/core/rest/node/field/Field.java [hunk 10]
```diff
@@ -11,4 +11,9 @@
 	 */
 	public String getType();
 
+	/**
+	 * Return the value stored in the field.
+	 * @return
+	 */
+	Object getValue();
```
Comment (Jotschi) [inline rest-model/src/main/java/com/gentics/mesh/core/rest/node/field/Field.java]: Check whether this impacts Jackson.

File: rest-model/src/main/java/com/gentics/mesh/core/rest/schema/change/impl/SchemaChangeOperation.java [hunk 11]
```diff
@@ -22,93 +12,6 @@
...
-	public String getAutoMigrationScript(Map<String, Object> properties) {
-		switch (this) {
-		case ADDFIELD:
-			return null;
-		case CHANGEFIELDTYPE:
-			String newType = String.valueOf(properties.get("type"));
-			if (newType != null) {
-				switch (newType) {
-				case "binary":
-					return loadAutoMigrationScript("typechange_binary.js");
```
Comment (Jotschi) [inline rest-model/src/main/java/com/gentics/mesh/core/rest/schema/change/impl/SchemaChangeOperation.java]: Check whether resource files have been removed.

File: core/src/test/java/com/gentics/mesh/core/schema/field/BooleanFieldMigrationTest.java (hunk 2 above)
Comment (philippguertler) [reply core/src/test/java/com/gentics/mesh/core/schema/field/BooleanFieldMigrationTest.java]: A custom script was used for renaming fields.

File: core/src/test/java/com/gentics/mesh/core/schema/field/BooleanListFieldMigrationTest.java (hunk 3 above)
Comment (philippguertler) [reply core/src/test/java/com/gentics/mesh/core/schema/field/BooleanListFieldMigrationTest.java]: A custom script was used for renaming fields.

File: core/src/test/java/com/gentics/mesh/core/schema/field/DateFieldMigrationTest.java (hunk 4 above)
Comment (philippguertler) [reply core/src/test/java/com/gentics/mesh/core/schema/field/DateFieldMigrationTest.java]: A custom script was used for renaming fields.

File: core/src/test/java/com/gentics/mesh/core/schema/field/DateListFieldMigrationTest.java (hunk 5 above)
Comment (philippguertler) [reply core/src/test/java/com/gentics/mesh/core/schema/field/DateListFieldMigrationTest.java]: A custom script was used for renaming fields.

File: core/src/test/java/com/gentics/mesh/core/schema/field/FieldMigrationTestcases.java (hunk 6 above)
Comment (philippguertler) [reply core/src/test/java/com/gentics/mesh/core/schema/field/FieldMigrationTestcases.java]: A custom script was used for renaming fields.

File: core/src/test/java/com/gentics/mesh/core/schema/field/HtmlFieldMigrationTest.java (hunk 7 above)
Comment (philippguertler) [reply core/src/test/java/com/gentics/mesh/core/schema/field/HtmlFieldMigrationTest.java]: A custom script was used for renaming fields.

**Comment by Jotschi** [comment]
buildnow

**Pull request merged**